lib/src/widgets/query.dart

```diff
@@ -62,22 +44,39 @@ class QueryState extends State<Query> {
       fetchResults: true,
       context: widget.options.context,
     );
+  }
+
+  void query(WatchQueryOptions options) {
+    final GraphQLClient client = GraphQLProvider.of(context).value;
+    assert(client != null);
+    observableQuery = client.watchQuery(options);
+  }
 
-    bool shouldCreateNewObservable = true;
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    query(_options);
+  }
 
-    if (observableQuery != null) {
-      if (observableQuery.options.areEqualTo(options)) {
-        shouldCreateNewObservable = false;
-      }
+  @override
+  void didUpdateWidget(Query oldWidget) {
+    super.didUpdateWidget(oldWidget);
 
-      await observableQuery.close();
-    }
+    final WatchQueryOptions newOptions = _options;
+
+    final bool shouldCreateNewObservable =
```

Comment:
Prefer to use `!observableQuery.options.areEqualTo(_options);` inside the if statement over two separate variable declarations.